src/stream/stream/filter_map.rs

```diff
@@ -31,13 +31,13 @@ where
         let this = self.project();
         let next = futures_core::ready!(this.stream.poll_next(cx));
         match next {
-            Some(v) => match (this.f)(v) {
-                Some(b) => Poll::Ready(Some(b)),
-                None => {
+            Some(v) => (this.f)(v).map_or_else(
```

Comment:
Was this clippy? I think this is harder to read.

Reply:
This was clippy's suggestion (or after a series of suggestions), removing unnecessary duplication